wkcuber/wkcuber/convert_raw.py

```diff
@@ -0,0 +1,239 @@
+    return indices
+
+
+def create_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "source_path",
+        help="Path to raw file to convert",
+        type=Path,
+    )
+
+    parser.add_argument(
+        "target_path", help="Output directory for the generated WKW dataset.", type=Path
+    )
+
+    parser.add_argument(
+        "--input_dtype",
+        help="Input dataset datatype (e.g. uint8, uint16, float32).",
+        required=True,
+    )
+
+    parser.add_argument(
+        "--shape",
+        help="Shape of the dataset (width, height, depth)",
+        type=parse_shape,
+        required=True,
+    )
+
+    add_scale_flag(parser, required=False)
+
+    parser.add_argument(
+        "--layer_name",
+        "-l",
+        help="Name of the cubed layer (color or segmentation).",
+        default="color",
+    )
+
+    parser.add_argument(
+        "--order",
+        help="The input data storage layout:"
+        "either 'F' for Fortran-style/column-major order (the default), "
+        "or 'C' for C-style/row-major order. "
+        "Note: Axes are expected in  (x, y, z) order.",
+        choices=("C", "F"),
+        default="F",
+    )
+
+    parser.add_argument(
+        "--flip_axes",
+        help="The axes which should be flipped. "
+        "Input format is a comma separated list of axis indices. "
+        "For example, 1,2,3 will flip the x, y and z axes.",
+        default=None,
+        type=parse_flip_axes,
+    )
+
+    parser.add_argument(
+        "--no_compress",
+        help="Don't compress saved data",
+        default=False,
+        action="store_true",
+    )
+
+    parser.add_argument(
+        "--max_mag",
+        "-m",
+        help="Max resolution to be downsampled. Needs to be a power of 2. In case of anisotropic downsampling, "
+        "the process is considered done when max(current_mag) >= max(max_mag) where max takes the "
+        "largest dimension of the mag tuple x, y, z. For example, a maximum mag value of 8 (or 8-8-8) "
+        "will stop the downsampling as soon as a magnification is produced for which one dimension is "
+        "equal or larger than 8. "
+        "The default value is calculated depending on the dataset size. In the lowest Mag, the size will be "
+        "smaller than 100vx per dimension",
+        type=int,
+        default=None,
+    )
+
+    add_sampling_mode_flag(parser)
+    add_interpolation_flag(parser)
+
+    add_distribution_flags(parser)
+
+    add_verbose_flag(parser)
+
+    return parser
+
+
+def _raw_chunk_converter(
+    bounding_box: BoundingBox,
+    source_raw_path: Path,
+    target_mag_view: MagView,
+    input_dtype: str,
+    shape: Tuple[int, int, int],
+    order: str,
+    flip_axes: Optional[Union[int, Tuple[int, ...]]],
+):
+    logging.info(f"Conversion of {bounding_box.topleft}")
+    cube_data = np.memmap(
```

Comment:
I learned from experiments that it is beneficial to copy the data in-memory, when the input order differs from the output order (wkw uses F). Otherwise, numpy may perform a lot of tiny reads, which is inefficient for IO, and memmapping doesn't always mitigate this. Not sure if copying also brings perf benefits in case the orders are the same.

Reply:
I just added a copy of the array.